tests/ut/azure_iot_adu_client_ut.c

```diff
@@ -745,6 +724,9 @@ static void testAzureIoTADUClient_SendAgentState_WithAgentStateAndRequestWithRet
     uint32_t ulRequestId;
     AzureIoTADUDeviceCustomProperties_t xDeviceCustomProperties;
 
+    /* We have to copy the expected payload to a scratch buffer since it's unescaped in place */
```

Comment:
Why can't this be left in-place?

Reply:
For the tests we need to. We have an expected payload, which we would like to reuse for different parsing scenarios. After the first test which uses the payload and parses it, the payload gets modified since it's unescaped in place. The next time we try and use that payload for a test, it no longer has an escaped JSON string, but instead has the unescaped in it's place, with a big gap until the end of the string. The JSON parsing then gets screwed up since it has unexpected characters.